src/lib/common/topology_sampler.cpp

```diff
@@ -64,53 +71,98 @@ static void topology_sample_cb(networkTopology_t* networkTopology) {
 
     // compile all additional info here
     _node_list.num_nodes = networkTopology->length;
-    configASSERT(sizeof(_node_list.nodes) >= _node_list.num_nodes * sizeof(uint64_t)); // if we hit this, expand the node list size
+    configASSERT(sizeof(_node_list.nodes) >=
+                 _node_list.num_nodes *
+                     sizeof(uint64_t)); // if we hit this, expand the node list size
 
-    network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.num_nodes), sizeof(_node_list.num_nodes));
+    network_crc32_calc = crc32_ieee_update(network_crc32_calc,
+                                           reinterpret_cast<uint8_t *>(&_node_list.num_nodes),
+                                           sizeof(_node_list.num_nodes));
 
     memset(_node_list.nodes, 0, sizeof(_node_list.nodes));
 
     neighborTableEntry_t *cursor = NULL;
     uint16_t counter;
-    for(cursor = networkTopology->front, counter = 0; (cursor != NULL) && (counter < _node_list.num_nodes); cursor = cursor->nextNode, counter++) {
+    xQueueReset(sys_info_crc_queue);
+    for (cursor = networkTopology->front, counter = 0;
+         (cursor != NULL) && (counter < _node_list.num_nodes);
+         cursor = cursor->nextNode, counter++) {
       _node_list.nodes[counter] = cursor->neighbor_table_reply->node_id;
-      network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.nodes[counter]), sizeof(uint64_t));
+      network_crc32_calc = crc32_ieee_update(
+          network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.nodes[counter]),
+          sizeof(uint64_t));
+      if (!sys_info_service_request(_node_list.nodes[counter], sys_info_reply_cb,
```

Comment:
After we have our list of nodes, now we can request the sys info from each node, which will come back async from `sys_info_reply_cb`